ESP32 Camera_web_server not compiling due to millis not declared.

### The problem

Been waiting for this one, thankyou.
Although it isn't compiling due to a millis not declared.

### Which version of ESPHome has the issue?

v2021.11.0-dev

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2021.11.2

### What platform are you using?

ESP32

### Board

m5stack-timer-cam and ESP32-cam

### Component causing the issue

esp32_camera_web_server

### Example YAML snippet

```yaml
esphome:
  name: m5-camera
  platform: ESP32
  board: m5stack-timer-cam
  
wifi:
  ssid: "*"
  password: "*"
 
# Enable logging
logger:

# Enable Home Assistant API
api:
  password: "*"

ota:
  password: "*"
  
esp32_camera:
  name: M5stack camera 2
  external_clock:
    pin: GPIO27
    frequency: 20MHz
  i2c_pins:
    sda: GPIO25
    scl: GPIO23
  data_pins: [GPIO17, GPIO35, GPIO34, GPIO5, GPIO39, GPIO18, GPIO36, GPIO19]
  vsync_pin: GPIO22
  href_pin: GPIO26
  pixel_clock_pin: GPIO21
  reset_pin: GPIO15
  
esp32_camera_web_server:
  - port: 8080
    mode: stream
  - port: 8081
    mode: stream
```


### Anything in the logs that might be useful for us?

```txt
NFO Reading configuration /config/esphome/m5-camera.yaml...
WARNING GPIO15 is a Strapping PIN and should be avoided.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Generating C++ source...
INFO Compiling app...
Processing m5-camera (board: m5stack-timer-cam; framework: arduino; platform: platformio/espressif32 @ 3.3.2)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
Dependency Graph
|-- <WiFi> 1.0
|-- <Update> 1.0
|-- <ESPmDNS> 1.0
Compiling /data/m5-camera/.pioenvs/m5-camera/src/esphome/components/esp32_camera_web_server/camera_web_server.cpp.o
Compiling /data/m5-camera/.pioenvs/m5-camera/src/main.cpp.o
src/esphome/components/esp32_camera_web_server/camera_web_server.cpp: In member function 'esp_err_t esphome::esp32_camera_web_server::CameraWebServer::streaming_handler_(httpd_req*)':
src/esphome/components/esp32_camera_web_server/camera_web_server.cpp:160:32: error: 'millis' was not declared in this scope
   uint32_t last_frame = millis();
                                ^
*** [/data/m5-camera/.pioenvs/m5-camera/src/esphome/components/esp32_camera_web_server/camera_web_server.cpp.o] Error 1
========================== [FAILED] Took 1.68 seconds ==========================
```


### Additional information

_No response_

--- a/esphome/components/esp32_camera_web_server/camera_web_server.cpp
+++ b/esphome/components/esp32_camera_web_server/camera_web_server.cpp
@@ -2,6 +2,7 @@
 
 #include "camera_web_server.h"
 #include "esphome/core/application.h"
+#include "esphome/core/hal.h"
 #include "esphome/core/helpers.h"
 #include "esphome/core/log.h"
 #include "esphome/core/util.h"

